code/modules/power/lighting.dm

```diff
@@ -420,8 +420,14 @@
 
 // called when area power state changes
 /obj/machinery/light/power_change()
-	spawn(10)
-		seton(powered())
+	. = ..()
+	if(.)
+		delay_and_set_on(expected_to_be_on(), 1 SECOND)
```

Comment:
I decoupled this interaction (power status changes, we turn lights on/off) from lightswitches turning lights on/off (they used to basically force the whole area to recompute power status, and then that would fall through to this proc and do the work).

I have no idea why this and the lightswitch interaction wait 1 second. If it's not desirable in one or both cases, I can easily make this happen immediately.

Reply:
Seems fine to me, gives a bit of tension, keeps behavior equivalent. Easy to remove if someone wants to down the track.

Reply:
iirc the wait is just to simulate neon lights powering up to match the sound effect.